doc/devel/coding_guide.rst

```diff
@@ -165,6 +165,48 @@ C/C++ extensions
 
 
 
+PR Review guidelines
+====================
+
+* We run on a 'use your judgement' system (for better or worse) and do
+  not have a strict nose-count rule.  If you have a commit bit, then
+  you are trusted to use it and to know when to ask for more review.
+
+* Do not self merge with the exception of un-breaking travis.
+
+* Squashing is case-by-case.  The balance is between burden on the
+  contributor, keeping a relatively clean history, and keeping a
+  history usable for bisecting.  The only time we are really strict
+  about it is to eliminate binary files (ex multiple test image
+  re-generations) and to remove upstream merges.
+
+* Be patient with new contributors.
+
+
+Backports
+=========
+
+
+When doing back ports please include the branch you backported the
+commit to along with the SHA in a comment on the original PR.
+
+A quick guide to how to back port a merge commit::
+
+  git remote update
+  git checkout target_branch
+  git merge --ff-only matplotlib/target_branch
+  git cherry-pick -m 1 TARGET_SHA
+  gitk   # to look at it
+  # local tests? (use your judgment)
+  git push DANGER target_branch
+  # leave a comment on PR noting sha of the resulting commit
+  # from the cherry-pick + branch it was moved to
+
+here matplotlib is a read-only remote to the matplotlib/matplotlib repo and DANGER in a read/write remote to the matplotlib/matplotlib repo.
```

Comment:
@anntzer The easiest solution is to use the https remote url. This way, you are always prompt for your username and password. It doesn't prevent you from actually pushing, but should raise red flags.